posthog-react-native/src/posthog-rn.ts

```diff
@@ -10,30 +10,31 @@ import {
 } from '../../posthog-core/src'
 import { PostHogMemoryStorage } from '../../posthog-core/src/storage-memory'
 import { getLegacyValues } from './legacy'
-import { SemiAsyncStorage } from './storage'
+import { SemiAsyncStorage, PostHogSemiAsyncStorage } from './storage'
 import { version } from './version'
 import { buildOptimisiticAsyncStorage, getAppProperties } from './native-deps'
-import { PostHogAutocaptureOptions, PostHogCustomAppProperties, PostHogCustomAsyncStorage } from './types'
+import { PostHogAutocaptureOptions, PostHogCustomAppProperties, PostHogStorage } from './types'
 import { withReactNativeNavigation } from './frameworks/wix-navigation'
 
 export type PostHogOptions = PosthogCoreOptions & {
+  /** If you use a customProvider that is synchronous, ie memory, JSI, cookies/localStorage on rn-web.
+   * You may `customSync` so posthog gets initialized with `new PostHog(..`
+   * */
   persistence?: 'memory' | 'file'
   /** Allows you to provide your own implementation of the common information about your App or a function to modify the default App properties generated  */
   customAppProperties?:
     | PostHogCustomAppProperties
     | ((properties: PostHogCustomAppProperties) => PostHogCustomAppProperties)
-  customAsyncStorage?: PostHogCustomAsyncStorage
+  customAsyncStorage?: PostHogStorage
```

Comment:
customAsyncStorage only needs to implement a subset of the posthog one, not sure if anyone would even use this